MANIFEST.in

```diff
@@ -0,0 +1,10 @@
+graft ops
+graft test
+
+include *.txt
+include *.md
+include MANIFEST.in
+include pyproject.toml
+include tox.ini
+
+global-exclude *~ *.py[cod] __pycache__ *.charm
```

Comment:
FWIW, I did a diff of the files in the 2.9.0 tar.gz from PyPI, and the 2.10.0 dev build locally. There's quite a bit of difference, mainly the test charms for the unit tests are now in it:

```diff
diff --git a/package.txt b/package.txt
index 8bdd835..c03bd35 100644
--- a/package.txt
+++ b/package.txt
@@ -1,5 +1,9 @@
 /
+/CHANGES.md
+/CODE_OF_CONDUCT.md
+/HACKING.md
 /LICENSE.txt
+/MANIFEST.in
 /PKG-INFO
 /README.md
 /ops/
@@ -29,8 +33,51 @@
 /ops.egg-info/top_level.txt
 /pyproject.toml
 /setup.cfg
-/setup.py
 /test/
+/test/__init__.py
+/test/bin/
+/test/bin/relation-ids
+/test/bin/relation-ids.bat
+/test/bin/relation-list
+/test/bin/relation-list.bat
+/test/charms/
+/test/charms/test_main/
+/test/charms/test_main/actions.yaml
+/test/charms/test_main/config.yaml
+/test/charms/test_main/lib/
+/test/charms/test_main/lib/__init__.py
+/test/charms/test_main/lib/ops/
+/test/charms/test_main/lib/ops/__init__.py
+/test/charms/test_main/lib/ops/_private/
+/test/charms/test_main/lib/ops/_private/__init__.py
+/test/charms/test_main/lib/ops/_private/timeconv.py
+/test/charms/test_main/lib/ops/_private/yaml.py
+/test/charms/test_main/lib/ops/charm.py
+/test/charms/test_main/lib/ops/framework.py
+/test/charms/test_main/lib/ops/jujuversion.py
+/test/charms/test_main/lib/ops/lib/
+/test/charms/test_main/lib/ops/lib/__init__.py
+/test/charms/test_main/lib/ops/log.py
+/test/charms/test_main/lib/ops/main.py
+/test/charms/test_main/lib/ops/model.py
+/test/charms/test_main/lib/ops/pebble.py
+/test/charms/test_main/lib/ops/py.typed
+/test/charms/test_main/lib/ops/storage.py
+/test/charms/test_main/lib/ops/testing.py
+/test/charms/test_main/lib/ops/version.py
+/test/charms/test_main/metadata.yaml
+/test/charms/test_main/src/
+/test/charms/test_main/src/charm.py
+/test/charms/test_smoke/
+/test/charms/test_smoke/README.md
+/test/charms/test_smoke/charmcraft.yaml
+/test/charms/test_smoke/metadata.yaml
+/test/charms/test_smoke/src/
+/test/charms/test_smoke/src/charm.py
+/test/fake_pebble.py
+/test/pebble_cli.py
+/test/smoke/
+/test/smoke/test_smoke.py
 /test/test_charm.py
 /test/test_framework.py
 /test/test_helpers.py
@@ -45,3 +92,4 @@
 /test/test_real_pebble.py
 /test/test_storage.py
 /test/test_testing.py
+/tox.ini
```

I think most of that stuff should be there to run the tests, right? Or should the .tar.gz not include tests at all?

Reply:
Ah, this is more like what I saw originally - maybe I did something incorrectly last time I did the comparison.

I feel like the extra *.md files should be there because the readme refers to them. The rest is the tests, yes. Personally, I think that the built package _should_ have everything that you need to run the tests (and that's a common pattern from what I can recall seeing). So I think this change is an improvement, because you have everything for the tests, instead of only some of the tests.

Reply:
Sounds reasonable to me, thanks!